### PR TITLE
core: Update mlir hash

### PR DESCRIPTION
**Pull request by math-fehr**
This PR updates the MLIR hash to point to the latest version.

### CODE REVIEW

**Review by superlopuh — APPROVED** [review]

File: tests/filecheck/mlir-conversion/with-mlir/dialects/llvm/llvm_pointer_ops.mlir [hunk 1]
```diff
@@ -16,10 +16,10 @@
 // CHECK: "builtin.module"() ({
 // CHECK-NEXT:   %0 = "arith.constant"() {"value" = 0 : i64} : () -> i64
 // CHECK-NEXT:   %1 = "llvm.inttoptr"(%0) : (i64) -> !llvm.ptr<i32>
-// CHECK-NEXT:   %2 = "llvm.load"(%1) : (!llvm.ptr<i32>) -> i32
+// CHECK-NEXT:   %2 = "llvm.load"(%1) {"ordering" = 0 : i64} : (!llvm.ptr<i32>) -> i32
```
Comment (webmiche) [inline tests/filecheck/mlir-conversion/with-mlir/dialects/llvm/llvm_pointer_ops.mlir:19]: What is this additional attribute here? Why do we not need to add it to the Load op? I guess it is just implicitly 0 if not  set?

**Review by webmiche — APPROVED** [review]

**Comment by PapyChacal** [comment]
Should we display the commit hash in the Readme? I would find it handy to recompile/compile for newcomers!

**Comment by PapyChacal** [comment]
If yes, to do after MLIR's compilation to have it cached already 🙂

File: tests/filecheck/mlir-conversion/with-mlir/dialects/llvm/llvm_pointer_ops.mlir (hunk 1 above)
Comment (math-fehr) [reply tests/filecheck/mlir-conversion/with-mlir/dialects/llvm/llvm_pointer_ops.mlir:19]: My guess is canonicalization?

**Review by AntonLydike — APPROVED** [review]
:bullettrain_side:

**Comment by webmiche** [comment]
> Should we display the commit hash in the Readme? I would find it handy to recompile/compile for newcomers!

We mention it in the interoperation markdown. Does that suffice or do would you prefer tighter integration?

**Comment by AntonLydike** [comment]
> We mention it in the interoperation markdown. Does that suffice or do would you prefer tighter integration?

I would put it in the README, in the install section, basically as a disclaimer "This version of xDSL is tested to work with MLIR commit <hash>, using other versions is not recommended and may lead to errors."

We stumbled upon that issue often enough now with newcomers that we want to be very upfront about the MLIR version.

File: tests/filecheck/mlir-conversion/with-mlir/dialects/mpi/mpi-hello-world-async.mlir [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// RUN: xdsl-opt %s -p lower-mpi | mlir-opt --convert-func-to-llvm --convert-memref-to-llvm --reconcile-unrealized-casts | filecheck %s
+// RUN: xdsl-opt %s -p lower-mpi | mlir-opt --convert-func-to-llvm --expand-strided-metadata --normalize-memrefs --memref-expand --fold-memref-alias-ops --finalize-memref-to-llvm --reconcile-unrealized-casts | filecheck %s
```
Comment (georgebisbas) [inline tests/filecheck/mlir-conversion/with-mlir/dialects/mpi/mpi-hello-world-async.mlir:1]: So all these are necessary due to the new hash?

Comment (math-fehr) [reply tests/filecheck/mlir-conversion/with-mlir/dialects/mpi/mpi-hello-world-async.mlir:1]: Yes, this is how `--convert-memref-to-llvm` is replaced in the new hash.

**Pull request merged**